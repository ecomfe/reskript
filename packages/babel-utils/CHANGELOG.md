# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.10.0](https://github.com/ecomfe/reskript/compare/v1.9.0...v1.10.0) (2021-07-20)


### Features

* **dev:** 增加一个babel插件为React组件注入对应源码路径 ([#91](https://github.com/ecomfe/reskript/issues/91)) ([fb0132d](https://github.com/ecomfe/reskript/commit/fb0132d31dc83128b8373da7b38ca0c3d32b4a9c))
