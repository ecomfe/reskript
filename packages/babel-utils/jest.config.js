module.exports = {
    preset: 'ts-jest/presets/js-with-ts',
    testMatch: ['**/__tests__/**/*.test.js'],
    transformIgnorePatterns: [
        '\\/dist\\/',
        'node_modules',
    ],
};
