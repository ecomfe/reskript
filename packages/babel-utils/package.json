{
  "name": "@reskript/babel-utils",
  "version": "1.10.0",
  "main": "dist/index.js",
  "license": "MIT",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "rm -rf dist && tsc",
    "test": "jest",
    "lint": "eslint src"
  },
  "devDependencies": {
    "@types/babel__core": "^7.1.15",
    "eslint": "^7.30.0",
    "jest": "^27.0.6",
    "ts-jest": "^27.0.3",
    "typescript": "^4.3.5"
  },
  "dependencies": {
    "@babel/core": "^7.14.6",
    "@babel/traverse": "^7.14.7"
  }
}
