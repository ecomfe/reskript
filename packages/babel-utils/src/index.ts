export * from './importReact';
export * from './testRunner';
export * from './component';
export * from './import';
export * from './traverse';
