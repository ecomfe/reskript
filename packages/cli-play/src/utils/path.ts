import fs from 'node:fs';
import path from 'node:path';

export const resolveComponentFileName = (componentModulePath: string): string => {
    const file = path.basename(componentModulePath, path.extname(componentModulePath));
    return file;
};

export const resolveComponentName = (componentModulePath: string): string => {
    const filename = resolveComponentFileName(componentModulePath);

    if (/[A-Z]/.test(filename)) {
        return filename;
    }

    return path.basename(path.dirname(componentModulePath));
};

export const resolveCasePath = (componentModulePath: string): string => {
    const directory = path.dirname(componentModulePath);
    const filename = resolveComponentFileName(componentModulePath);
    const caseFileName = path.join(directory, '__repl__', `${filename}.case.md`);
    return caseFileName;
};

const PLAY_CONFIG_EXTENSIONS = ['.ts', '.tsx', '.js', '.jsx'];

export const resolveLocalConfigurationPath = (componentModulePath: string): string | undefined => {
    const directory = path.dirname(componentModulePath);
    const filename = resolveComponentFileName(componentModulePath);
    const files = PLAY_CONFIG_EXTENSIONS.map(v => path.join(directory, '__repl__', `${filename}.play${v}`));
    return files.find(fs.existsSync);
};
