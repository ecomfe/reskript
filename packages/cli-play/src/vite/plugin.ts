import fs from 'node:fs/promises';
import {Plugin} from 'vite';
import {buildEntryScript, resolveEntryPath, BuildEntryOptions} from '../utils/entry.js';

interface Options extends BuildEntryOptions {
    enableConcurrentMode: boolean;
}

export default function playEntryPlugin(options: Options): Plugin {
    return {
        name: 'reskript:play-entry',
        enforce: 'pre',
        resolveId(id) {
            if (id === '/playground-entry.jsx') {
                return id;
            }
        },
        async load(id) {
            if (id === '/playground-entry.jsx') {
                const content = await fs.readFile(resolveEntryPath(options.enableConcurrentMode), 'utf-8');
                return buildEntryScript(content, options);
            }
        },
    };
}
