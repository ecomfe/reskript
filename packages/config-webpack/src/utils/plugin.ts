import {Compilation, Compiler} from 'webpack';
import HtmlWebpackPlugin from 'html-webpack-plugin';

export class TransformHtmlWebpackPlugin {
    readonly transform: (html: string) => string;

    constructor(transform: (html: string) => string) {
        this.transform = transform;
    }

    apply(compiler: Compiler) {
        compiler.hooks.compilation.tap('transform-html-webpack-plugin', compilation => this.inject(compilation));
    }

    inject(compilation: Compilation) {
        const {afterTemplateExecution} = HtmlWebpackPlugin.getHooks(compilation);
        afterTemplateExecution.tap(
            'transform-html-webpack-plugin',
            data => ({...data, html: this.transform(data.html)})
        );
    }
}
