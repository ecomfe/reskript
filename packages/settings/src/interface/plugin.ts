import {CommandInput} from './command.js';
import {ProjectSettings} from './project.js';

type SettingsPluginReturn = ProjectSettings | Promise<ProjectSettings>;

export type SettingsPluginItem = (current: ProjectSettings, cmd: CommandInput) => SettingsPluginReturn;

export type MayBeSettingsPlugin = SettingsPluginItem | null | undefined | false;

export type SettingsPlugin = MayBeSettingsPlugin | MayBeSettingsPlugin[];
