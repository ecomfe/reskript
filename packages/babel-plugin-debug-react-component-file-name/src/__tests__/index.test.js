import path from 'path';
import {createTestRunner} from '@reskript/babel-utils';
import plugin from '../index';

const runSuite = createTestRunner(
    __dirname,
    {test, expect},
    {
        plugins: [
            [
                plugin,
                {srcDirectory: path.resolve(__dirname, 'fixtures', 'src')},
            ],
        ],
    }
);

runSuite();
