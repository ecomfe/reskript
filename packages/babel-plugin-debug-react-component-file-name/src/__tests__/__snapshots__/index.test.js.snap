// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export default component 1`] = `
"import useComponentFile from \\"@reskript/babel-plugin-debug-react-component-file-name/useComponentFile\\";
// export default component
export default function Foo() {
  useComponentFile(\\"export-default-component.js\\")
  return 'foo';
}"
`;

exports[`multiple components 1`] = `
"import useComponentFile from \\"@reskript/babel-plugin-debug-react-component-file-name/useComponentFile\\";

// multiple components
function Foo() {
  useComponentFile(\\"multiple-components.js\\")
  return 'foo';
}

function Bar() {
  useComponentFile(\\"multiple-components.js\\")
  return 'bar';
}"
`;

exports[`no component 1`] = `
"// no component
function foo() {
  return 'foo';
}"
`;

exports[`outside src directory 1`] = `
"// outside src directory
export default function Foo() {
  return 'foo';
}"
`;
